server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -544,4 +550,11 @@ public void sendHeartbeat() {
         shuffleServer.getStorageManager().getStorageInfo(),
         shuffleServer.getNettyPort());
   }
+
+  public boolean isActivateClientPartitionReassign() {
+    if (decommissionActivateClientReassignEnabled && serverStatus.get() != ServerStatus.ACTIVE) {
```

Comment:
The `ActivateClientPartitionReassign` state should not be dependent on `ServerStatus`, as the trigger for partition reassignment logic comes from the client.

Reply:
I think you misunderstand this patch. This is for client reassign when server is healthy or in decommission

Reply:
My mistake. get it.